dashboard/src/components/DeploymentFormBody/BasicDeploymentForm/Param.tsx

```diff
@@ -72,7 +74,14 @@ export default function Param({
     paramDeploymentEvent?: DeploymentEvent,
   ): boolean => {
     if (paramDeploymentEvent == null) {
-      return getValue(appValues, path) === (expectedValue ?? true);
+      let val = getValue(appValues, path);
+      // retrieve the value that the property pointed by path should have to be hidden.
+      // https://github.com/kubeapps/kubeapps/issues/1913
+      if (val === undefined) {
+        const target = otherParams.find(p => p.path === path);
```

Comment:
mmh, just noticed that the params are not a flat array but it has nested params (for the case of a subsection). You need to add some recursion here, if not this would only detect top-level params.